src/Frame.cpp

```diff
@@ -836,13 +840,13 @@ void Frame::AddImage(std::shared_ptr<QImage> new_image, bool only_odd_lines)
 				start = 1;
 
 			for (int row = start; row < image->height(); row += 2) {
-				memcpy((unsigned char *) pixels, new_pixels + (row * image->bytesPerLine()), image->bytesPerLine());
-				new_pixels += image->bytesPerLine();
+				int offset = row * image->bytesPerLine();
+				memcpy(pixels + offset, new_pixels + offset, image->bytesPerLine());
```

Comment:
This is where things really break.

The way that loop was previously written, it would go through every other row (odd or even) in `new_pixels`, and copy that row's data **to the first row of `pixels`** — every time! Nothing was ever offsetting the `pixels` location to advance it by the `row` stride, and nothing would have ever written to any part of `pixels` (and therefore `image`) beyond the first scanline/row.

Reply:
(Oh, yeah — the telltale cast on `pixels` — to cast away its `const` status, which it shouldn't have had in the first place — is also fun, in the previous code.)